From c390da81844a82aa5903f1477f1332ba6f7fd6a2 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sat, 25 Jul 2026 21:38:32 +0900
Subject: [PATCH 09/10] Add -lmachuser -lhurduser to libc.so on GNU/Hurd.

http://lists.gnu.org/archive/html/bug-hurd/2011-03/msg00112.html

2011-03-29  Samuel Thibault  <samuel.thibault@ens-lyon.org>

        * Makerules ($(inst_libdir)/libc.so): Add -lmachuser -lhurduser to
        libc.so on GNU/Hurd.

It's still unclear what we want to aim for.
---
 Makerules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makerules b/Makerules
index 6bef57ece9..a50032b7f4 100644
--- a/Makerules
+++ b/Makerules
@@ -1026,6 +1026,9 @@ $(inst_libdir)/libc.so: $(common-objpfx)format.lds \
 	      '$(libdir)/$(patsubst %,$(libtype.oS),$(libprefix)$(libc-name))'\
 	      ' AS_NEEDED (' $(rtlddir)/$(rtld-installed-name) ') )' \
 	) > $@.new
+ifeq ($(patsubst gnu%,,$(config-os)),)
+	echo 'INPUT ( AS_NEEDED ( -lmachuser -lhurduser ) )' >> $@.new
+endif
 	mv -f $@.new $@
 
 endif
-- 
2.54.0

